jerry-core/api/jerry.c

```diff
@@ -447,6 +447,26 @@ jerry_eval (const jerry_char_t *source_p, /**< source code */
 } /* jerry_eval */
 
 /**
+ * Run enqueued Promise jobs until the first thrown error or until all get executed.
+ *
+ * Note:
+ *      returned value must be freed with jerry_release_value, when it is no longer needed.
+ *
+ * @return result of last executed job, may be error value.
+ */
+jerry_value_t
+jerry_run_all_enqueued_jobs (void)
```

Comment:
Yes, it just like the following code in nodejs, which will lead to infinite loop, but it is not engine/runtime's fault.
```
function foo()
{
  console.log("1");
  process.nextTick(foo);
}
foo();
```